rust/agents/validator/src/server/eigen_node.rs

```diff
@@ -0,0 +1,272 @@
+use axum::{
+    http::StatusCode,
+    response::IntoResponse,
+    routing::{get, Router},
+    Json,
+};
+use derive_new::new;
+use hyperlane_base::CoreMetrics;
+use hyperlane_core::HyperlaneDomain;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+enum ServiceStatus {
+    Up,
+    Down,
+    Initializing,
+}
+
+// struct for body of /node
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+struct NodeInfo {
+    node_name: String,
+    spec_version: String,
+    node_version: String,
+}
+
+// struct for body of /node/services
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+struct Service {
+    id: String,
+    name: String,
+    description: String,
+    status: ServiceStatus,
+}
+
+/// A server that serves EigenLayer specific routes
+#[derive(new)]
+pub struct EigenNodeAPI {
+    origin_chain: HyperlaneDomain,
+    core_metrics: Arc<CoreMetrics>,
+}
+
+impl EigenNodeAPI {
+    /// Function to create the eigen_node_router
+    pub fn router(&self) -> Router {
+        let core_metrics_clone = self.core_metrics.clone();
+        let origin_chain = self.origin_chain.clone();
+
+        tracing::info!("Serving the EigenNodeAPI routes...");
+
+        Router::new().nest(
+            "/node",
+            Router::new()
+                .route(
+                    "/health",
+                    get(move || Self::node_health_handler(origin_chain, core_metrics_clone)),
+                )
+                .nest(
```

Comment:
This places the services endpoint at `/node/health/services` I think, which differs from the spec https://eigen.nethermind.io/docs/spec/api/#get-eigennodeservices

You should move the router instantiations above, in their own variables, to minimize nesting here - that makes it pretty hard to read. Iirc 3 levels of indentation is the max you should aim for in a function (at least per the linux kernel [style guide](https://www.kernel.org/doc/html/v4.10/process/coding-style.html?highlight=indentation#indentation))

Reply:
I don't think so. The test for ./node/services works fine.

But yes, you're right. It ca separated into it's own variables.